clang/labs/lab1/kulaev_zhenya/AttrubuteAlways.cpp

```diff
@@ -0,0 +1,58 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/Attr.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+using namespace clang;
+
+class AlwaysInlineVisitor : public RecursiveASTVisitor<AlwaysInlineVisitor> {
+public:
+  AlwaysInlineVisitor(ASTContext *myContext) : myContext(myContext) {}
+
+  bool VisitFunctionDecl(FunctionDecl *func) {
+    Stmt *Body = func->getBody();
+    if (Body && isa<CompoundStmt>(Body)) {
+      CompoundStmt *BodyCompound = cast<CompoundStmt>(Body);
+      for (Stmt *S : BodyCompound->body()) {
+        if (!isa<IfStmt>(S)) {
+          func->addAttr(AlwaysInlineAttr::CreateImplicit(*myContext));
+          llvm::outs() << "__attribute__((always_inline)) "
+                       << func->getNameAsString() << "\n";
+        }
+      }
+    }
+    return true;
+  }
+
+private:
+  ASTContext *myContext;
+};
+
+class AlwaysInlineConsumer : public ASTConsumer {
+public:
+  AlwaysInlineConsumer(ASTContext *myContext) : myVisitor(myContext) {}
+
+  void HandleTranslationUnit(ASTContext &myContext) override {
+    myVisitor.TraverseDecl(myContext.getTranslationUnitDecl());
+  }
+
+private:
+  AlwaysInlineVisitor myVisitor;
+};
+
+class AlwaysInlinePlugin : public PluginASTAction {
+public:
+  std::unique_ptr<clang::ASTConsumer>
+  CreateASTConsumer(CompilerInstance &Compiler,
+                    llvm::StringRef) override {
+    return std::make_unique<AlwaysInlineConsumer>(&Compiler.getASTContext());
+  }
+  bool ParseArgs(const CompilerInstance &Compiler,
+                 const std::vector<std::string> &args) override {
```

Comment:
Could you please fix warnings printed by clang-tidy test: https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8250927820/job/22566738976?pr=6

Reply:
There are still some warnings in clang-tidy that can be fixed.

Reply:
We will fix it)